src/AppInstallerCLICore/Workflows/UriValidationFlow.cpp

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "UriValidationFlow.h"
+#include <AppInstallerDownloader.h>
+#include <UriValidation/UriValidation.h>
+#include <regex>
+
+namespace AppInstaller::CLI::Workflow
+{
+    // Check if smart screen is required for a given zone.
+    bool IsSmartScreenRequired(Settings::SecurityZoneOptions zone)
+    {
+        auto isSmartScreenEnabled = Settings::GroupPolicies().IsEnabled(Settings::TogglePolicy::Policy::SmartScreenCheck);
+        AICLI_LOG(Core, Info, << "SmartScreen validation is " << (isSmartScreenEnabled ? "enabled" : "disabled"));
+
+        auto isSecurityZoneCheckRequired = zone == Settings::SecurityZoneOptions::Internet || zone == Settings::SecurityZoneOptions::UntrustedSites;
+        AICLI_LOG(Core, Info, << "Security zone check is " << (isSecurityZoneCheckRequired ? "required" : "not required"));
+
+        return isSmartScreenEnabled && isSecurityZoneCheckRequired;
+    }
+
+    // Check if the given uri is blocked by smart screen.
+    bool IsBlockedBySmartScreen(Execution::Context& context, const std::string& uri)
+    {
+        auto response = AppInstaller::UriValidation::ValidateUri(uri);
+        switch (response.Decision())
+        {
+        case AppInstaller::UriValidation::UriValidationDecision::Block:
+            AICLI_LOG(Core, Error, << "URI '" << uri << "' was blocked by smart screen. Feedback URL: " << response.Feedback());
+            context.Reporter.Error() << Resource::String::UriBlockedBySmartScreen << std::endl;
+            return true;
+        case AppInstaller::UriValidation::UriValidationDecision::Allow:
+        default:
+            return false;
+        }
+    }
+
+    // Get Uri zone for a given uri or file path.
+    HRESULT GetUriZone(const std::string& uri, Settings::SecurityZoneOptions* zone)
+    {
+        if (!zone)

```

Comment:
Probably better to pass as a reference (`Settings::SecurityZoneOptions& zone`) so that you don't have to worry about bad pointers